pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -115,6 +129,74 @@ func (qt *QueueTracker) decreaseTrackedResource(queuePath string, applicationID
 	return removeQT, nil
 }
 
+func (qt *QueueTracker) setMaxApplications(count uint64, queuePath string) error {
+	log.Logger().Debug("Setting max applications",
+		zap.String("queue path", queuePath),
+		zap.Uint64("max applications", count))
+	var childQueuePath, immediateChildQueueName string
+	childQueuePath, immediateChildQueueName = getChildQueuePath(queuePath)
+	childQueueTracker := qt
+	if childQueuePath != "" {
+		for childQueuePath != "" {
+			if childQueueTracker != nil {
+				if len(childQueueTracker.childQueueTrackers) == 0 || childQueueTracker.childQueueTrackers[immediateChildQueueName] == nil {
+					newChildQt := newQueueTracker(immediateChildQueueName)
+					childQueueTracker.childQueueTrackers[immediateChildQueueName] = newChildQt
+					childQueueTracker = newChildQt
+				} else {
+					childQueueTracker = childQueueTracker.childQueueTrackers[immediateChildQueueName]
+				}
+			}
+			childQueuePath, immediateChildQueueName = getChildQueuePath(childQueuePath)
+		}
+	}
+	if childQueueTracker.maxRunningApps != 0 && count != 0 && len(childQueueTracker.runningApplications) > int(count) {
+		log.Logger().Warn("Current running applications is greater than config max applications",
+			zap.String("queue path", queuePath),
+			zap.Uint64("current max applications", childQueueTracker.maxRunningApps),
+			zap.Int("total running applications", len(childQueueTracker.runningApplications)),
+			zap.Uint64("config max applications", count))
+		return fmt.Errorf("current running applications is greater than config max applications for %s", queuePath)
+	} else {
+		childQueueTracker.maxRunningApps = count
+	}
+	return nil
+}
+
+func (qt *QueueTracker) setMaxResources(resource *resources.Resource, queuePath string) error {
```

Comment:
This method looks almost the same as `setMaxApplications()`, there are a lot of common things, could you extract it? Plus, there's missing coverage here, please address that.